campaigntools.php

```diff
@@ -14,6 +14,52 @@ function campaigntools_civicrm_pageRun(&$page) {
   }
 }
 
+/**
+ * Implements hook_civicrm_buildForm().
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_buildForm
+ */
+function campaigntools_civicrm_buildForm($formName, &$form) {
+  // Detect contribution page and contribution confirm page
+  if (
+      $formName === 'CRM_Contribute_Form_Contribution_Main'
+      || $formName === 'CRM_Contribute_Form_Contribution_Confirm'
+      || $formName === 'CRM_Event_Form_Registration_Register'
+      || $formName === 'CRM_Event_Form_Registration_Confirm'
+    ) {
+    // List all campaign ids available
+    $campaign = array(
+      'ids' => array(
+        1, 2,
+      ),
+    );
+    // Get the campaign id by the entryURL in the controler array
+    $controller = $form->getVar('controller');
+    // Process to get the campaign id in the parameter
+    $params = explode('?', $controller->_entryURL);
+    parse_str(end($params), $parseURL);
+    $paramItems = [];
+
+    // Remove amp; since it was not remove using parse_str
+    foreach($parseURL as $key => $value) {
+      $newKey = str_replace('amp;', '', $key);
+      $paramItems[$newKey] = $value;
+    }
+
+    // Check if campaign id is set and match it on the ones in campaign_ids
+    if (isset($paramItems['campaign']) && in_array($paramItems['campaign'], $campaign['ids'])) {
```

Comment:
Rather than using hard-coded campaign IDs in https://github.com/twomice/com.joineryhq.campaigntools/pull/2/files#diff-fc91c7d4b3b44f8e7169e2dcc43f32ccR31, and then using `in_array()` here, please use the `campaign.getCount` api to test whether a campaign with this ID exists. You can try this first in the api explorer (https://[my-devsite.example.com]/civicrm/api3#explorer) to see what's returned from that api with both valid and invalid campaign IDs